lib/ace/lib/event.js

```diff
@@ -115,6 +115,29 @@ exports.capture = function(el, eventHandler, releaseCaptureHandler) {
     return onMouseUp;
 };
 
+exports.addTouchMoveListener = function (el, callback) {
+    if ("ontouchmove" in el) {
+        var startx, starty;
+        exports.addListener(el, "touchstart", function (e) {
+            var touchObj = e.changedTouches[0];
+            startx = parseInt(touchObj.clientX);
+            starty = parseInt(touchObj.clientY);
+        });
+        exports.addListener(el, "touchmove", function (e) {
+            var factor = 1,
+            touchObj = e.changedTouches[0];
+
+            e.wheelX = -(parseInt(touchObj.clientX) - startx) / factor;
```

Comment:
long and int are the same in javascript.
Some browsers can also return fractional values for clientX, but they support fractional values for things like style.top, and work better if scroll values are not rounded.